### PR TITLE
ShardOwner interface

### PR DESCRIPTION
**Pull request by jshencode**
use ShardOwner interface to remove hard coded shards

during query time, if query specified shards, use query specified shards for query
if query does not specify shards, use owned shards

### CODE REVIEW

File: broker/executor.go [hunk 1]
```diff
@@ -56,7 +56,7 @@ func (qe *queryExecutorImpl) Execute(ctx context.Context, sqlQuery string, w htt
 	qc := &query.AQLQueryContext{
 		Query: aqlQuery,
 	}
-	qc.Compile(qe.tableSchemaReader)
+	qc.Compile(qe.tableSchemaReader, topology.NewTopologyShardOwner(qe.topo))
```
Comment (shz117) [inline broker/executor.go]: heads up: we won't need this after my PR merged

Comment (jshencode) [reply broker/executor.go]: ok. can remove that during the merge

File: datanode/datanode.go [hunk 2]
```diff
@@ -469,7 +474,18 @@ func (d *dataNode) AssignShardSet(shardSet shard.ShardSet) {
 			}
 		}()
 	}
-	d.RUnlock()
+}
+
+// GetOwnedShards returns all shard ids the datanode owns
+func (d *dataNode) GetOwnedShards() []int {
```
Comment (voyager-dw) [inline datanode/datanode.go]: kind of duplicate as above (t *topologyShardOwner) GetOwnedShards, can we use the d.topo directkly?

Comment (jshencode) [reply datanode/datanode.go]: no topologyShardOwner gets all shards from the topology

this get the shards this host owns

**Pull request merged**